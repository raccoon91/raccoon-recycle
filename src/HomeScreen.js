import React, { Component } from 'react';
import { StyleSheet, Text, View, Button, Image, TouchableOpacity } from 'react-native';
import * as Expo from 'expo';
import { connect } from 'react-redux';
import { loginUserName } from './actions';

class HomeScreen extends Component {
  state = {
    signedIn: false,
    name: '',
    photoUrl: ''
  };

  static navigationOptions = {
    title: 'Home'
  };

  signIn = async () => {
    try {
      const result = await Expo.Google.logInAsync({
        androidClientId: '1050254961075-9f9osk0h0kvc562l3sh0pbhc5bvv7ift.apps.googleusercontent.com',
        scopes: ['profile', 'email']
      });

      if (result.type === 'success') {
        this.setState({
          signedIn: true,
          photoUrl: result.user.photoUrl
        });

        this.props.loginUserName(result.user.name);
      } else {
        console.log('cancelled');
      }
    } catch (err) {
      console.log('error', err);
    }
  }

  render() {
    console.log(this.state.photoUrl);

    return (
      <View style={styles.container}>
        {
          this.state.signedIn
          ? <LoggedInPage name={this.props.username} photoUrl={this.state.photoUrl} navigation={this.props.navigation} />
          : <LoginPage signIn={this.signIn} />
        }
      </View>
    );
  }
}

const LoginPage = props => {
  return (
    <View style={{ alignItems: 'center' }}>
      <View style={{ width: 180, height: 180, alignItems: 'center', justifyContent: 'center', backgroundColor: '#a7c0cd', marginBottom: 140, borderRadius: 20 }}>
        <Image
          style={{ width: 140, height: 140 }}
          source={require('../assets/recycle.png')}
        />
      </View>
      <TouchableOpacity style={{ flexDirection: 'row', alignItems:'center', backgroundColor: '#4b636e', padding: 5, borderRadius: 5 }} onPress={() => props.signIn()}>
        <View style={{ width: 50, height: 50, alignItems: 'center', justifyContent: 'center', backgroundColor: '#ffffff', borderRadius: 5 }}>
          <Image
            style={{ width: 40, height: 40 }}
            source={require('../assets/google_logo.png')}
          />
        </View>
        <Text style={{ fontSize: 22, fontWeight:'bold', color: '#ffffff', marginHorizontal: 20 }}>Sign in with Google</Text>
      </TouchableOpacity>
    </View>
  );
};

const LoggedInPage = props => {
  return (
    <View style={styles.container}>
      <Text style={{ color: 'white', fontSize: 34, fontWeight: 'bold', marginBottom: 10 }}>Welcome!</Text>
      <Text style={{ color: 'white', fontSize: 28, fontWeight: 'bold', marginBottom: 20 }}>{props.name}</Text>
      <Image style={styles.image} source={{ uri: props.photoUrl }} />
      <TouchableOpacity style={{ backgroundColor: '#4b636e', marginTop: 100, paddingHorizontal: 20, paddingVertical: 10, borderRadius: 5 }} onPress={() => props.navigation.navigate('BarcodeScanner')}>
        <Text style={{ fontSize: 20, color: '#ffffff', fontWeight: 'bold' }}>Click To Scan</Text>
      </TouchableOpacity>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#78909c',
    alignItems: 'center',
    justifyContent: 'center',
  },
  image: {
    marginTop: 15,
    width: 100,
    height: 100,
    borderColor: "rgba(0,0,0,0.2)",
    borderWidth: 3,
    borderRadius: 150
  }
});

function mapStateToProps(state) {
  return {
    username: state.userName
  };
};

function mapDispatchToProps(dispatch) {
  return {
    loginUserName: (name) => {
      dispatch(loginUserName(name));
    }
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(HomeScreen);