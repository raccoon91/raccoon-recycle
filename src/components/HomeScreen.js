import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Google } from 'expo';
import PropTypes from 'prop-types';
import {
  StyleSheet,
  Text,
  View,
  Image,
  TouchableOpacity,
} from 'react-native';
import { loginUserName } from '../actions';

class HomeScreen extends Component {
  static navigationOptions = {
    header: null
  };

  state = {
    signedIn: false,
    photoUrl: ''
  };

  signIn = async () => {
    try {
      // development
      // const clientId = '1050254961075-9f9osk0h0kvc562l3sh0pbhc5bvv7ift.apps.googleusercontent.com';

      // deploy
      const clientId = '1050254961075-qruht9q5sdrjgg5l0k521q4pt9e2dgq6.apps.googleusercontent.com';
      const { type, user } = await Google.logInAsync({ clientId });
      const { saveloginUserName } = this.props;

      if (type === 'success') {
        this.setState({
          signedIn: true,
          photoUrl: user.photoUrl
        });

        saveloginUserName(user.name);
      }
    } catch (err) {
      console.log('error', err);
    }
  }

  render() {
    const { signedIn, photoUrl } = this.state;
    const { username, navigation } = this.props;

    return (
      <View style={styles.container}>
        {
          signedIn
            ? <LoggedInPage name={username} photoUrl={photoUrl} navigation={navigation} />
            : <LogInPage signIn={this.signIn} />
        }
      </View>
    );
  }
}

const LogInPage = ({ signIn }) => (
  <View style={{ alignItems: 'center' }}>
    <View style={styles.appLogoContainer}>
      <Image
        style={{ width: 140, height: 140 }}
        source={require('../../assets/images/recycle.png')}
      />
    </View>
    <TouchableOpacity style={styles.loginButton} onPress={() => signIn()}>
      <View style={styles.googleLogoWrapper}>
        <Image
          style={{ width: 40, height: 40 }}
          source={require('../../assets/images/google_logo.png')}
        />
      </View>
      <Text style={styles.loginText}>Login in with Google</Text>
    </TouchableOpacity>
  </View>
);

const LoggedInPage = ({ name, photoUrl, navigation }) => (
  <View style={styles.container}>
    <Text style={styles.logedinText}>Welcome!</Text>
    <Text style={styles.userName}>{name}</Text>
    <Image style={styles.userImage} source={{ uri: photoUrl }} />
    <TouchableOpacity style={styles.scanButton} onPress={() => navigation.navigate('BarcodeScanner')}>
      <Text style={styles.scanText}>Click To Scan</Text>
    </TouchableOpacity>
  </View>
);

HomeScreen.propTypes = {
  username: PropTypes.string.isRequired,
  navigation: PropTypes.objectOf(Object).isRequired,
  saveloginUserName: PropTypes.func.isRequired
};

LogInPage.propTypes = {
  signIn: PropTypes.func.isRequired
};

LoggedInPage.propTypes = {
  name: PropTypes.string.isRequired,
  photoUrl: PropTypes.string.isRequired,
  navigation: PropTypes.objectOf(Object).isRequired
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#78909c'
  },
  appLogoContainer: {
    justifyContent: 'center',
    alignItems: 'center',
    marginTop: 30,
    marginBottom: 140,
    width: 180,
    height: 180,
    borderRadius: 20,
    backgroundColor: '#a7c0cd'
  },
  loginButton: {
    flexDirection: 'row',
    alignItems: 'center',
    padding: 5,
    borderRadius: 5,
    backgroundColor: '#4b636e'
  },
  googleLogoWrapper: {
    justifyContent: 'center',
    alignItems: 'center',
    width: 50,
    height: 50,
    borderRadius: 5,
    backgroundColor: '#ffffff'
  },
  loginText: {
    marginHorizontal: 20,
    color: '#ffffff',
    fontSize: 22,
    fontWeight: 'bold'
  },
  logedinText: {
    marginBottom: 10,
    color: 'white',
    fontSize: 34,
    fontWeight: 'bold'
  },
  userName: {
    marginBottom: 20,
    color: 'white',
    fontSize: 28,
    fontWeight: 'bold'
  },
  userImage: {
    marginTop: 15,
    width: 100,
    height: 100,
    borderColor: 'rgba(0,0,0,0.2)',
    borderWidth: 3,
    borderRadius: 150
  },
  scanButton: {
    paddingVertical: 10,
    paddingHorizontal: 20,
    marginTop: 100,
    borderRadius: 5,
    backgroundColor: '#4b636e'
  },
  scanText: {
    color: '#ffffff',
    fontSize: 20,
    fontWeight: 'bold'
  }
});

const mapStateToProps = state => ({
  username: state.userName
});

const mapDispatchToProps = dispatch => ({
  saveloginUserName: (name) => {
    dispatch(loginUserName(name));
  }
});

export default connect(mapStateToProps, mapDispatchToProps)(HomeScreen);
