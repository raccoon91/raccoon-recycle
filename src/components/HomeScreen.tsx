import React from 'react';
import { connect } from 'react-redux';
import { Dispatch } from 'redux';
import { Google } from 'expo';
import {
  StyleSheet,
  Text,
  View,
  Image,
  TouchableOpacity,
} from 'react-native';
import { loginUserName, Action } from '../actions';
import { StoreState } from '../types';

export interface Props {
  username: string;
  saveloginUserName: (name: string) => void;
  navigation: {
    navigate: (route: string) => void
  };
}

export interface LogInPageProps {
  signIn: () => void;
}

export interface LoggedInPageProps {
  name: string;
  photoUrl: string;
  navigation: {
    navigate: (route: string) => void
  };
}

interface State {
  signedIn: boolean;
  photoUrl: string;
}

class HomeScreen extends React.Component<Props, State> {
  static navigationOptions = {
    header: null
  };

  state = {
    signedIn: false,
    photoUrl: ''
  };

  signIn = async () => {
    try {
      // development
      const clientId = '1050254961075-9f9osk0h0kvc562l3sh0pbhc5bvv7ift.apps.googleusercontent.com';

      // deploy
      // const clientId = '1050254961075-qruht9q5sdrjgg5l0k521q4pt9e2dgq6.apps.googleusercontent.com';
      const { type, user } = await Google.logInAsync({ clientId });
      const { saveloginUserName } = this.props;

      if (type === 'success') {
        this.setState({
          signedIn: true,
          photoUrl: user.photoUrl
        });

        saveloginUserName(user.name);
      }
    } catch (err) {
      console.log('error', err);
    }
  }

  render() {
    const { signedIn, photoUrl } = this.state;
    const { username, navigation } = this.props;

    return (
      <View style={styles.container}>
        {
          signedIn
            ? <LoggedInPage name={username} photoUrl={photoUrl} navigation={navigation} />
            : <LogInPage signIn={this.signIn} />
        }
      </View>
    );
  }
}

const LogInPage = ({ signIn }: LogInPageProps) => (
  <View style={{ alignItems: 'center' }}>
    <View style={styles.appLogoContainer}>
      <Image
        style={{ width: 140, height: 140 }}
        source={require('../../assets/images/recycle.png')}
      />
    </View>
    <TouchableOpacity style={styles.loginButton} onPress={() => signIn()}>
      <View style={styles.googleLogoWrapper}>
        <Image
          style={{ width: 40, height: 40 }}
          source={require('../../assets/images/google_logo.png')}
        />
      </View>
      <Text style={styles.loginText}>Login in with Google</Text>
    </TouchableOpacity>
  </View>
);

const LoggedInPage = ({ name, photoUrl, navigation }: LoggedInPageProps) => (
  <View style={styles.container}>
    <Text style={styles.logedinText}>Welcome!</Text>
    <Text style={styles.userName}>{name}</Text>
    <Image style={styles.userImage} source={{ uri: photoUrl }} />
    <TouchableOpacity style={styles.scanButton} onPress={() => navigation.navigate('BarcodeScanner')}>
      <Text style={styles.scanText}>Click To Scan</Text>
    </TouchableOpacity>
  </View>
);

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#78909c'
  },
  appLogoContainer: {
    justifyContent: 'center',
    alignItems: 'center',
    marginTop: 30,
    marginBottom: 140,
    width: 180,
    height: 180,
    borderRadius: 20,
    backgroundColor: '#a7c0cd'
  },
  loginButton: {
    flexDirection: 'row',
    alignItems: 'center',
    padding: 5,
    borderRadius: 5,
    backgroundColor: '#4b636e'
  },
  googleLogoWrapper: {
    justifyContent: 'center',
    alignItems: 'center',
    width: 50,
    height: 50,
    borderRadius: 5,
    backgroundColor: '#ffffff'
  },
  loginText: {
    marginHorizontal: 20,
    color: '#ffffff',
    fontSize: 22,
    fontWeight: 'bold'
  },
  logedinText: {
    marginBottom: 10,
    color: 'white',
    fontSize: 34,
    fontWeight: 'bold'
  },
  userName: {
    marginBottom: 20,
    color: 'white',
    fontSize: 28,
    fontWeight: 'bold'
  },
  userImage: {
    marginTop: 15,
    width: 100,
    height: 100,
    borderColor: 'rgba(0,0,0,0.2)',
    borderWidth: 3,
    borderRadius: 150
  },
  scanButton: {
    paddingVertical: 10,
    paddingHorizontal: 20,
    marginTop: 100,
    borderRadius: 5,
    backgroundColor: '#4b636e'
  },
  scanText: {
    color: '#ffffff',
    fontSize: 20,
    fontWeight: 'bold'
  }
});

const mapStateToProps = (state: StoreState) => ({
  username: state.userName
});

const mapDispatchToProps = (dispatch: Dispatch<Action>) => ({
  saveloginUserName: (name: string) => {
    dispatch(loginUserName(name));
  }
});

export default connect(mapStateToProps, mapDispatchToProps)(HomeScreen);
