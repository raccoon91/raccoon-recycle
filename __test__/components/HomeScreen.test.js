import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import HomeScreen from '../../src/HomeScreen';

describe('<HomeScreen />', () => {
  let component = null;

  it('render correctly', () => {
    component = shallow(<HomeScreen />);
  });

  it('matches snapshot', () => {
    expect(component).toMatchSnapshot();
  })
})