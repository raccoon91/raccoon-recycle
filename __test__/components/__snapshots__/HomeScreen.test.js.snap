// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeScreen /> matches snapshot 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
